pkg/model/utils.go

```diff
@@ -202,3 +203,24 @@ func getListDiff(l1, l2 []string) []string {
 	}
 	return added
 }
+
+func SanitizeName(name string) (string, error) {
+	name = strings.TrimSpace(name)
+	name = strings.ToLower(name)
+
+	name = ValidKubeNameRegex.ReplaceAllString(name, "-")
+	name = strings.TrimSuffix(name, "-")
+	name = strings.TrimPrefix(name, "-")
+
+	// if multiple "-" replace all for only one
+	re, err := regexp.Compile(`-(-+)`)
+	if err != nil {
+		return "", err
+	}
+	name = re.ReplaceAllString(name, "-")
+	// trim the repository name for internal use in labels
+	if len(name) > 63 {
+		return name[:63], nil
```

Comment:
this might leave and ending `_` again

Reply:
`_` are replaced by `-` with the kube validation, but it might get a `-` indeed, leet me review this